Tools/clinic/clinic.py

```diff
@@ -4617,15 +4617,18 @@ def parse(self, block: Block) -> None:
                 fail("'preserve' only works for blocks that don't produce any output!")
             block.output = self.saved_output
 
-    @staticmethod
-    def valid_line(line: str) -> bool:
+    def in_docstring(self) -> bool:
+        """Return true if we are processing a docstring."""
+        return "docstring" in self.state.__name__
```

Comment:
Perhaps this is a tad too hacky.

Reply:
```suggestion
        return self.state in {self.state_parameter_docstring,
                              self.state_function_docstring}
```

Reply:
> Perhaps this is a tad too hacky.

uhhh yes :)

The current PR's implementation of `in_docstring()` returns `True` if `self.state` is `state_parameter_docstring`, `state_function_docstring` or `state_parameter_docstring_start`. Your suggested change only accounts for two of those states. Which is correct? :)

Reply:
Yes, we only want this in the parsing states, not the docstring-start state :)